src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
So how to continue? It will be also okay for me, if we expose the two objects directly and remove the CreateContainerCmd#withLink and other HostConfig delegating commends. I can make this changes in this pull requests, if you like?